pkg/core/resource_graph.go

```diff
@@ -20,8 +20,8 @@ func (cg *ResourceGraph) AddResource(resource Resource) {
 	cg.underlying.AddVertex(resource)
 }
 
-func (cg *ResourceGraph) AddDependency(source Resource, dest Resource) {
-	cg.underlying.AddEdge(source.Id(), dest.Id())
+func (cg *ResourceGraph) AddDependency(thingThatIsDependedOn Resource, thingThatDepends Resource) {
```

Comment:
while these are a bit clearer its still a bit hard on the eyes lol

Reply:
I'll revert this, since it conflicts anyway with #449